### PR TITLE
Bug regex changes in front-end to make it similar regex in back-end

### PR DESCRIPTION
**Pull request by saif-ericsson**
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/resources/static/js/subscription.js [hunk 1]
```diff
@@ -861,9 +861,15 @@ jQuery(document).ready(function () {
             $('#subscriptionNameInput').addClass("is-invalid");
             error = true;
         }
-
+        
+        // When this regExp need to be changed then remember to change the one in the
+        // back-end (invalidSubscriptionNameRegex in SubscriptionValidator.java), which do the same
+        // invalid subscription name check. The two
+        // regEx always need to be the same for ensuring the same check.
         // /(\W)/ Is a regex that matches anything that is not [A-Z,a-z,0-8] and _.
```
Comment (tobiasake) [inline src/main/resources/static/js/subscription.js]: Shouldn't it be possible to use dash "-" in SubscriptionName also, or is it only underscore?

File: src/main/resources/static/js/subscription.js [hunk 2]
```diff
@@ -896,6 +902,10 @@ jQuery(document).ready(function () {
             var emails = notificationMeta.split(",");
             for(var email of emails) {
                 email = email.trim();
+                // When this regExp need to be changed then remember to change the one in the
+                // back-end (validEmailRegExpression in SubscriptionValidator.java), which do the same
+                // email validation check. The two
+                // regEx always need to be the same for ensuring the same check.
```
Comment (tobiasake) [inline src/main/resources/static/js/subscription.js]: Manual test some mail addresses that should succeed and some mail addresses that should fail before merge.

File: src/main/resources/static/js/subscription.js (hunk 1 above)
Comment (saif-ericsson) [reply src/main/resources/static/js/subscription.js]: This regEx doesnt accept dash (-) in the subscription name... as we discussed, it shouldnt be in the name.

Comment (tobiasake) [reply src/main/resources/static/js/subscription.js]: Ok,,, I just trying to think of what and which subscription name variants that user want to use.

**Pull request merged**